integration/auth/user_mgt_test.go

```diff
@@ -368,37 +368,221 @@ func TestCreateUser(t *testing.T) {
+		providerIDs = mapToProviderIDs(userRecord.ProviderUserInfo)
+		if contains(providerUIDs, googleFederatedUID) {
+			t.Errorf("UpdateUser().ProviderUserInfo[*].UID want NOT include %v; got %v",
+				googleFederatedUID, providerUIDs)
+		}
+		if contains(providerIDs, "google.com") {
+			t.Errorf("UpdateUser().ProviderUserInfo[*].ProviderID want NOT include 'google.com'; got %v",
+				providerIDs)
+		}
+	})
+
+	t.Run("UnlinkMultipleProvidersAtOnce", func(t *testing.T) {
+		deletePhoneNumberUser(t, "+15555550001")
+
+		googleFederatedUID := "google_uid_" + generateRandomAlphaNumericString(10)
+		facebookFederatedUID := "facebook_uid_" + generateRandomAlphaNumericString(10)
+
+		userRecord, err := client.UpdateUser(context.Background(), updateUser.UID,
+			(&auth.UserToUpdate{}).
+				PhoneNumber("+15555550001").
+				ProviderToLink((&auth.UserProvider{
+					ProviderID: "google.com",
+					UID:        googleFederatedUID,
+				})))
+		if err != nil {
+			t.Fatal(err)
+		}
+		userRecord, err = client.UpdateUser(context.Background(), updateUser.UID,
+			(&auth.UserToUpdate{}).
+				ProviderToLink((&auth.UserProvider{
+					ProviderID: "facebook.com",
+					UID:        facebookFederatedUID,
+				})))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		providerUIDs := mapToProviderUIDs(userRecord.ProviderUserInfo)
+		providerIDs := mapToProviderIDs(userRecord.ProviderUserInfo)
+		wantAll := []string{googleFederatedUID, facebookFederatedUID, "+15555550001"}
+		if !containsAll(providerUIDs, wantAll) {
+			t.Errorf("UpdateUser().ProviderUserInfo[*].UID want include all %v; got %v",
+				wantAll, providerUIDs)
+		}
+		wantAll = []string{"google.com", "facebook.com", "phone"}
+		if !containsAll(providerIDs, wantAll) {
+			t.Errorf("UpdateUser().ProviderUserInfo[*].ProviderID want include all %v; got %v",
+				wantAll, providerIDs)
+		}
+
+		userRecord, err = client.UpdateUser(context.Background(), updateUser.UID,
+			(&auth.UserToUpdate{}).
+				ProvidersToDelete([]string{"google.com", "facebook.com", "phone"}))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		providerUIDs = mapToProviderUIDs(userRecord.ProviderUserInfo)
+		providerIDs = mapToProviderIDs(userRecord.ProviderUserInfo)
+		notWantAll := []string{googleFederatedUID, facebookFederatedUID, "+15555550001"}
+		if !containsNone(providerUIDs, notWantAll) {
+			t.Errorf("UpdateUser().ProviderUserInfo[*].UID want not include all %v; got %v",
+				notWantAll, providerUIDs)
+		}
+		notWantAll = []string{"google.com", "facebook.com", "phone"}
+		if !containsNone(providerIDs, notWantAll) {
+			t.Errorf("UpdateUser().ProviderUserInfo[*].ProviderID want not include all %v; got %v",
+				notWantAll, providerIDs)
+		}
+	})
+
+	t.Run("NoopsGivenEmptyProvidersToDelete", func(t *testing.T) {
+		userRecord := createTestUser("NoopWithEmptyProvidersToDeleteUser")
+		defer deleteUser(userRecord.UID)
+
+		gotUserRecord, err := client.UpdateUser(context.Background(), userRecord.UID,
```

Comment:
Shouldn't this fail with a validation error. We got:

```
        if len(u.params) == 0 {
		// update without any parameters is never allowed
		return nil, fmt.Errorf("update parameters must not be nil or empty")
	}
```

Perhaps we should change `ProvidersToDelete()` function ignore empty array, if the existing value for that key is already empty?